src/sentry/api/event_search.py

```diff
@@ -626,7 +626,10 @@ def get_snuba_query_args(query=None, params=None):
             if snuba_name in ('start', 'end'):
                 kwargs[snuba_name] = term.value.value
             elif snuba_name in ('project_id', 'issue'):
-                kwargs['filter_keys'][snuba_name] = term.value.value
+                value = term.value.value
+                if isinstance(value, int):
+                    value = [value]
+                kwargs['filter_keys'][snuba_name] = value
```

Comment:
Oh, I just saw the case in https://github.com/getsentry/sentry/pull/13553.

I think this is actually kind of a bug with passing a project id or issue id via a `query` rather than querystring params. If you had a query like `issue.id:1 issue.id:2` then it would just use the last issue id we parsed.

Above, change
```
kwargs = {
   ....
   'filter_keys': {},
}
```
to
```
kwargs = {
   ....
   'filter_keys': defaultdict(list),
}
```

Then change this line to:
`kwargs['filter_keys'][snuba_name].extend(value)`.
That way we always include all project ids, regardless if from querystring or query

Reply:
Perfect - updated ❤️